.eleventy.js

```diff
@@ -29,22 +22,15 @@ module.exports = function (eleventyConfig) {
     eleventyConfig.addPassthroughCopy({"manifest.json": "manifest.json"});
     eleventyConfig.addPassthroughCopy({"src/js": "js"});
 
+    eleventyConfig.addPassthroughCopy({
+        "node_modules/d3/dist/d3.min.js": "lib/d3.min.js",
+        "node_modules/d3-cloud/build/d3.layout.cloud.js": "lib/d3.layout.cloud.js"
+    });
+
     // Configure BrowserSync.
     eleventyConfig.setBrowserSyncConfig({
         ...eleventyConfig.browserSyncConfig,
-        callbacks: {
-            ready: (error, browserSync) => {
-                // TODO: Add custom 404 page.
-                const content404 = fs.readFileSync("dist/404.html");
-
-                // Provides the 404 content without redirect.
-                browserSync.addMiddleware("*", (request, response) => {
-                    response.write(content404);
-                    response.writeHead(404);
-                    response.end();
-                });
-            }
-        }
```

Comment:
This section of code ensures 404.html is shown when accessing an non-existent page. When it's removed, accessing an non-existent page shows an error page with an unfriendly message - `Cannot GET {pageUrl}`.

Reply:
I don't understand why this material is integrated with the browserSync configuration rather than at top level as part of either netlify or eleventy's. The most appropriate audience for this friendly message would be the production configuration rather than only as part of the dev preview environment in which browserSync is enabled

Reply:
You are right, @amb26. Netlify supports [this feature](https://docs.netlify.com/routing/redirects/redirect-options/#custom-404-page-handling) that the redirection to 404 page happens automatically for all paths that don’t resolve to a static file. It doesn’t require any redirect rules.